parser_library/src/processing/preprocessors/cics_preprocessor.cpp

```diff
@@ -941,47 +937,60 @@ class cics_preprocessor final : public preprocessor
 
     bool try_asm_xopts(std::string_view input, size_t lineno)
     {
-        if (input.substr(0, 5) != "*ASM ")
+        static constexpr auto sv_icase_compare = [](const std::string_view& sv_a, const std::string_view& sv_b) {
+            if (sv_a.size() != sv_b.size())
+                return false;
+
+            return std::equal(
+                sv_a.begin(), sv_a.end(), sv_b.begin(), [](auto a, auto b) { return tolower(a) == tolower(b); });
+        };
+
+        if (!sv_icase_compare(input.substr(0, 5), "*ASM "))
```

Comment:
I think this was one of the exceptions and it is case sensitive (only the *ASM part)